sdk/src/main/java/com/google/cloud/dataflow/sdk/util/common/Counter.java

```diff
@@ -418,17 +430,44 @@ public boolean isCompatibleWith(Counter<?> that) {
 
   //////////////////////////////////////////////////////////////////////////////
 
-  /** The name of this counter. */
-  protected final String name;
-
   /** The kind of aggregation function to apply to this counter. */
   protected final AggregationKind kind;
 
   /** The commit state of this counter. **/
   protected final AtomicReference<CommitState> commitState;
 
+  /** A hashable name structure which will uniquely identify a counter. */
+  public interface Name {
+    public String name();
+    @Nullable public String systemName();
```

Comment:
Calling these "name" is a bit odd. Could we call them "contextSystemName" or "contextNameSystem"?